tonic/src/codec/decode.rs

```diff
@@ -254,9 +255,18 @@ impl StreamingInner {
             None => None,
         };
 
-        Poll::Ready(if let Some(data) = chunk {
-            self.buf.put(data);
-            Ok(Some(()))
+        Poll::Ready(if let Some(frame) = chunk {
+            match frame {
+                frame if frame.is_data() => {
+                    self.buf.put(frame.into_data().unwrap());
+                    Ok(Some(()))
+                }
+                frame if frame.is_trailers() => {
+                    self.trailers = Some(frame.into_trailers().unwrap());
```

Comment:
Do we want to potentially merge the trailers rather than replace the old ones? This sounds like another lack of guarantee from the Body trait that isn't super clear what we should do. I think the safe bet is to merge?

Reply:
I think that makes sense - I can't find any specific documentation on what the semantics of `poll_frame` should be. The `Body` trait vaugely implies that trailers should only come at the end, but even in that case it might still be legal to end a body with two trailer frames.

Reply:
@seanmonstar do you have any context on this? I'd be happy to submit a docs PR for `http_body` if you can clarify the intended semantics?

Reply:
There _should_ only be one trailers frame. That's what h2 and h3 do. But some things are overly complicated to encode into the type system... So, I guess someone _could_ make a body that does weird things.